public/app/plugins/datasource/tempo/SearchTraceQLEditor/TraceQLSearch.tsx

```diff
@@ -85,37 +88,47 @@ const TraceQLSearch = ({ datasource, query, onChange }: Props) => {
     fetchTags();
   }, [datasource]);
 
+  useEffect(() => {
+    // Initialize state with configured static filters that already have a value from the config
+    datasource.search?.filters
+      ?.filter((f) => f.value)
+      .forEach((f) => {
+        if (!findFilter(f.id)) {
+          updateFilter(f);
```

Comment:
This does not seem right to me. As I understand it we add filters from the config if they are not at the moment in the UI. This makes sense but then we do not remove them when the data source changes. So if I start with one data source this adds it's static fields, then I switch to other one with other static fields I end up with UI that will show static fields from both data sources?

Reply:
Not exactly sure how to fix this but I still get to a state like this if I have 2 data sources with fields setup:
![Screenshot from 2023-03-28 16-28-56](https://user-images.githubusercontent.com/1014802/228271491-9913f2e3-d353-451c-ae26-ebd31204a4d4.png)

I think we just don't clean the query state when changing data source, which may have some value if the data source is the same type so makes sense to keep the query. I am not really sure how the defaults should work in this case. Do we even need the default values?

Reply:
I like the idea. Use case is "I'm an admin who owns configuration of this datasource and want to save my end users a few clicks by suggesting the value that I anticipate they'll use for filtering in 80%+ of cases." 

That being said, I haven't had any customer feedback one way or another yet. We can ask Paymentsense about this next week when they take this for a drive? I can't tell from this thread how buggy or not you're saying this interaction is - would they be able to test what you're got in this draft PR or are the bugs identified here such that you don't want to merge it at all? Would be nice if they can take it for a drive on their own data and make the determination of if it'd be useful, but if not, I can always ask them about it conceptually. 

Btw, I'd actually be interested to probe how much they find the ability to set default values useful versus the ability to require that the user apply some specific set of filters (i.e. force the user to choose some non-empty value for the static fields). This is more about enforcing good querying practice and less about saving the user clicks/giving them breadcrumbs in-product though.

Reply:
So the main thing is I am not sure how this should work in this particular case.

We preserve query if we move between the same data sources, like switching between Tempo-dev and Tempo-prod. In case of these default fields, should they be preserved? If so should they be merged into the defaults of the other data source somehow? Or override them? Should they be treated as part of the query from the point they are visible or should they be part of the query only if user changes them and only then override/merge with the other data source?

Reply:
@adrapereira I also just now realized that we encode this in the query here: https://github.com/grafana/grafana/blob/bfb0dde4a8b5d297a341ab44846ec9c78b5ea1ef/public/app/plugins/datasource/tempo/dataquery.gen.ts#L91. I am not sure how is this supposed to work if the query should work with different data source? Also if the datasource definition of these fields change should that affect the saved queries in dashboard for example? I think this information should no be encoded in the query as that ties is to the data source config.